crypto/ocsp/ocsp_test.cc

```diff
@@ -623,3 +790,87 @@ TEST(OCSPTest, TestRevokedOCSP) {
   ASSERT_EQ(1, ocsp_resp_find_status_res);
   ASSERT_EQ(V_OCSP_CERTSTATUS_REVOKED, status);
 }
+
+// Test valid OCSP response, but the data itself is untrusted
+TEST(OCSPTest, TestUntrustedDataOCSP) {
+  // convert const good ocsp response test file to changeable pointer
+  bssl::Span<const uint8_t> der = bssl::Span<const uint8_t>(ocsp_response_der);
+  std::unique_ptr<uint8_t[]> ptr(new uint8_t[der.size()]);
+  OPENSSL_memcpy(ptr.get(), der.data(), der.size());
+
+  // flip a byte right in the middle of the cert
+  ptr[800] = ptr[800] + 1;
```

Comment:
I copied this test from [s2n's unit tests](https://github.com/aws/s2n-tls/blob/main/tests/unit/s2n_x509_validator_test.c#L1034-L1082). 
Basically what it's trying to test is that the OCSP Response itself can be parsed because the OCSP related fields are still correct, but the certificate within the OCSP Response has been malformed. So I think random bytes and changing the data would both work in this test case.

Reply:
Thanks for explanation.

Reply:
The response is signed. This test flips a byte (somewhere in the data that is signed). The resulting signature verification should fail and the function should report back that the data is untrusted. Which is why you input `0` in your call to `ExtractAndVerifyBasicOCSP` below and not `1` (return value of `1` from `OCSP_basic_verify` indicates that the signature verified successfully). Could potentially make that particular argument to `ExtractAndVerifyBasicOCSP` more explicit e.g. define a macro for the expected value.

It is not because the response is necessarily malformed. Imagine changing the response status from `unauthorized` to `successful`. It is not malformed, but should not verify.

Reply:
Added macros for the expected status value 👍